src/plugins/timeseries-plugin.ts

```diff
@@ -0,0 +1,258 @@
+import { OffscreenRenderEngine } from '../engines/offscreen-render-engine';
+import { SeparatedInteractionsEngine } from '../engines/separated-interactions-engine';
+import { HitRegion, RegionTypes } from '../types';
+import UIPlugin from './ui-plugin';
+
+export type TimeseriesPoint = [number, number];
+
+interface TimeseriesPointsSummary {
+    min: number;
+    max: number;
+    first: number;
+    last: number;
+}
+
+export type TimeseriesPluginStyles = {
+    height?: number;
+    padding?: number;
+    color?: string;
+};
+
+type TimeseriesPluginStylesNoOptional = {
+    height: number;
+    padding: number;
+    color: string;
+};
+
+export const defaultCPUPluginStyles: TimeseriesPluginStylesNoOptional = {
+    height: 68,
+    padding: 5,
+    color: 'red',
+};
+
+type TimeseriesPointWithIndex = [idx: number, ts: number, value: number, normalizedTs: number, normalizedValue: number];
+
+enum TimeseriesPointWithIndexX {
```

Comment:
It a very useless enum. It greatly complicates the code. Better to do without it.

Reply:
The enum was to make the code more readable - to stop hardcode numberic indexes; [0] is less readable than [Enum.timestamp]

If you feel strongly - i can remove it - but the reason i introduced it was i had bugs with my indexes

Reply:
Do you want me to use object instead? therefore not needing the index enum at all?

Reply:
You can just go the other way - do not create data for yourself on each render, but use the data from `this.data` each time. https://github.com/pyatyispyatil/flame-chart-js/pull/44#discussion_r1064683060